song_translator/track/views.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework import generics, permissions
+from rest_framework.response import Response
+from google_trans_new import google_translator
+
+from .models import Track, Singer
+from .serializers import SingerSerializer, TrackSerializer
+from .permisisions import IsOwnerOrReadOnly
+
+
+class TrackList(generics.ListCreateAPIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    queryset = Track.objects.all()
+    serializer_class = TrackSerializer
+
+
+class TrackDetail(generics.RetrieveUpdateDestroyAPIView):
+    queryset = Track.objects.all()
+    serializer_class = TrackSerializer
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly]
+
+
+class TrackTranslate(generics.RetrieveUpdateDestroyAPIView):
+
+    queryset = Track.objects.all()
+    serializer_class = TrackSerializer
+
+    def get(self, request, pk, lang):
+        translator = google_translator()
+        return Response(translator.translate(self.get_object().text, lang_tgt=lang))
```

Comment:
hmm, that means we are translating song every time on issuing GET request, doesn't sound good to me. Let's rather translate at the moment of Track creation and save the translation to the database.